segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/DefragmentOperation.java

```diff
@@ -144,11 +147,28 @@
                             CompletableFuture<Void> f;
                             if (chunksToConcat.size() > 1) {
                                 // Concat
-                                f = concatChunks();
+                                f = concatChunks()
+                                .handleAsync((vv, ex) -> {
+                                    if (null != ex) {
+                                        ex = Exceptions.unwrap(ex);
+                                        if (ex instanceof InvalidOffsetException) {
+                                            // Skip ahead by 1 chunk.
+                                            targetChunkName = chunksToConcat.get(1).getName();
+                                            chunksToConcat.clear();
+                                            skipFailed.set(true);
+                                            return null;
```

Comment:
Do we need to add logs reporting that this exception has been captured? Otherwise, how are we going to realize that the defragmentation process is getting `InvalidOffsetException`?

Reply:
Good suggestion. I'll add logs.

Reply:
added